x/valset-pref/simulation/sim_msgs.go

```diff
@@ -0,0 +1,112 @@
+package simulation
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	osmosimtypes "github.com/osmosis-labs/osmosis/v13/simulation/simtypes"
+
+	valsetkeeper "github.com/osmosis-labs/osmosis/v13/x/valset-pref"
+	"github.com/osmosis-labs/osmosis/v13/x/valset-pref/types"
+)
+
+func RandomMsgSetValSetPreference(k valsetkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*types.MsgSetValidatorSetPreference, error) {
+	// Generate random digit from 0-1 with
+	//randWeight := RandomWeight()
```

Comment:
note: i tried to simulate with random weights from 0-1 but the possibilities were endless, hence all the  test was failing. Therefore, I hard coded the weight to be 1

Reply:
Wouldn't another way be to do this as follows:

Start with a weight of 1
Generate a random number of validators
For each validator, generate a random number between [0,remainingweight)
With that number, subtract from remaining weight
Last validator in the sequence simply gets the remaining

Does that make sense?

Reply:
this works! thank you :))